go-app-ussd_pmtct_seed.js

```diff
@@ -58,30 +58,109 @@ go.app = function() {
+                        'pilot config: ' + JSON.stringify(pilot_config),
+                    ].join('\n'));
+                })
+                .then(function () {
+                    // NOTE:
+                    //      If we're able to tell from local state what channel is supposed to be
+                    //      used then use that rather than the HTTP call.
+                    //      Because of how Seed's services work using asynchronous webhooks there
+                    //      can be a race condition if we check the subscriptions too soon after
+                    //      creating a new registration
+                    console.log(self.im.user.answers.state_register_pmtct);
+                    if(self.im.user.answers.state_create_pmtct_registration === 'whatsapp') {
+                        return pilot_config.channel;
+                    } else {
+                        return self.im.config.services.message_sender.channel;
+                    }
+
+                    return sbm
+                        .is_identity_subscribed(self.im.user.answers.identity.id, [/whatsapp/])
+                        .then(function(confirmed) {
+                            if(confirmed) {
+                                return pilot_config.channel;
+                            } else {
+                                return self.im.config.services.message_sender.channel;
+                            }
+                        });
+                })
+                .then(function(channel) {
+                    return self.im
+                        .log('Returning channel ' + channel + ' for ' + self.im.user.addr)
+                        .then(function() {
+                            return channel;
+                        });
+                });
+        };
+
         self.send_registration_thanks = function() {
-            return ms
-            .create_outbound_message(
-                self.im.user.answers.identity.id,
-                self.im.user.answers.msisdn,
-                self.im.user.i18n($(
-                    "HIV positive moms can have an HIV negative baby! You can get free " +
-                    "medicine at the clinic to protect your baby and improve your health"
-                ))
-            )
-            .then(function() {
-                return ms
-                .create_outbound_message(
-                    self.im.user.answers.identity.id,
-                    self.im.user.answers.msisdn,
-                    self.im.user.i18n($(
-                        "Recently tested HIV positive? You are not alone, many other pregnant " +
-                        "women go through this. Visit b-wise.mobi or call the AIDS Helpline " +
-                        "0800 012 322"
-                    ))
-                );
-            });
+            return self
+                .get_channel()
+                .then(function(channel) {
+                    console.log(channel);
```

Comment:
Is this log intended to be here or is it for debugging?

Reply:
No, it was for debugging. I removed it from the source file, but didn't update the built file. https://github.com/praekeltfoundation/ndoh-jsbox/pull/132/commits/dc2782cb23f7ac834074c02c130878e46eac7c4d removes this.